internal/advtools/convo.lua

```diff
@@ -61,8 +61,111 @@ local function generateKeywordsForChoice(choice)
     addKeywords(choice, new_keywords)
 end
 
+-- Helper function to create new dialog choices, returns the created choice
+local function new_choice(choice_type, title, keywords)
+    local choice = df.adventure_conversation_choice_infost:new()
+    choice.cc = df.talk_choice:new()
+    choice.cc.type = choice_type
+    local text = df.new("string")
+    text.value = title
+    choice.print_string.text:insert("#", text)
+
+    if keywords ~= nil then
+        addKeywords(choice, keywords)
+    end
+    return choice
+end
+
+local function addHistFigWhereaboutsChoice(profile)
+    for i, c in pairs(adventure.conversation.conv_choice_info) do
+        if c.cc.type == df.talk_choice_type.AskWhereabouts and
+          c.cc.invocation_target_hfid ~= -1 and
+          c.cc.invocation_target_hfid == profile.histfig_id then
+            -- Don't add repeat entries
+            return
+        end
+    end
+
+    local histfig = df.historical_figure.find(profile.histfig_id)
+    local name = ""
+    local creature = df.creature_raw.find(histfig.race)
+    if creature then
+        local caste = creature.caste[histfig.caste]
+        name = caste.caste_name[0]
+    end
+    local title = "Ask for the whereabouts of the " .. name .. " " .. dfhack.TranslateName(histfig.name, true)
```

Comment:
do we only need the english name here? are they never referenced by their native name?

Reply:
In vanilla convo system yeah they're only referenced by their English name

Reply:
is that what you *want*, though? @Ozzatron what do you think?